client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -18,28 +22,88 @@ import type { IAppState } from 'calypso/state/types';
 interface Props {
 	planSlugs: PlanSlug[];
 	withoutProRatedCredits?: boolean;
+	storageAddOns?: ( AddOnMeta | null )[] | null;
+	currencyCode?: string | null;
+}
+
+/***
+ * The 2023 Pricing Grid handles prices differently than add-ons, so for the time being we need to
+ * do some conversion to calculate totals correctly.
+ *
+ * The 2023 pricing grid currently handles prices in some currencies in their smallest unit, like
+ * the Japanese Yen. For others, however, like USD, prices are not handled in their smallest unit,
+ * like the US dollar.
+ *
+ * For example, 2000 yen would be represented by the integer 2000 since a yen is the smallest unit.
+ * $20.00, however, would be represented as the integer 20, and not 2000. Since storage add-on
+ * prices are always handled in the smallest unit, we need to convert them, when appropriate, to the
+ * correct integer.
+ *
+ * We should expose a more friendly function from the format-currency package
+ */
+function convertPriceForSmallestUnit( price: number, currencyCode: string | null ) {
+	const currencyObject = getCurrencyObject( price, currencyCode || '', {
+		isSmallestUnit: true,
+	} );
+	return parseInt( currencyObject.integer.replace( /,/g, '' ) );
+}
+
+function getTotalPrices( planPrices: PlanPrices, addOnPrice = 0 ): PlanPrices {
+	const totalPrices = { ...planPrices };
+	let key: keyof PlanPrices;
+
+	for ( key in totalPrices ) {
+		const price = totalPrices[ key ];
+
+		if ( ! ( price === null ) ) {
+			// Display prices are rounded to the nearest dollar,
+			// so we do the same for the add-on price
+			totalPrices[ key ] = price + Math.round( addOnPrice );
```

Comment:
I see that here you want to round to the "nearest dollar", by which I suppose you mean rounding to the nearest whole unit in the currency's most common unit. I understand that this would not work if we switch to using integer pricing, but can you explain what this is for? It seems to me that significantly changing prices on the front-end is not the right place to do it.

Reply:
Thanks for the input! I believe I cover the challenge here

https://github.com/Automattic/wp-calypso/pull/81338#discussion_r1318929055